src/Datasource/EntityTrait.php

```diff
@@ -1090,6 +1115,29 @@ protected function _nestedErrors($field)
         return [];
     }
 
+    /**
+     * Reads if there are errors for one or many objects.
+     *
+     * @param array|\Cake\Datasource\EntityTrait $object The object to read errors from.
```

Comment:
The param type should be `mixed`. Since you are passing all properties to this method, `$object` can be of any type.